bot.py

```diff
@@ -595,12 +601,49 @@ def get_members_by_role(role):
             santa = naughtylist.pop(0)
             child = naughtylist.pop(0)
             if santa == message.author:
-                print(santa.name + "\n" + child.name)
                 msg = "Some Santa you are, " + santa.name + ". You forgot about " + child.name + "."
+                log.info(msg)
                 await client.send_message(message.author, msg)
                 return
         #Their name was not found
         msg = "Looks like you fucked up, " + message.author.name + ", your name isn't on the list.\nBetter luck next year!"
         await client.send_message(message.author, msg)
 
+    #Send reminder DMs to all participants
+    elif message.content.lower().startswith("!noticemesanta"):
+        log.info("[{0}] requested mass reminder".format(member))
+        authorized_users = [
+            "115183069555589125", # aceat64
+            "221162619497611274", # HungryNinja
+            "130586766754316288"  # PeasAndClams
+        ]
+        if member.id not in authorized_users:
+            await client.send_file(message.channel, 'dennis.gif')
+            msg = "Just what do you think you're doing, cucko?\nYou're not the boss of me."
+            await client.send_message(message.channel, msg)
+            return
+        try:
+            with open(os.path.join(sys.path[0], 'naughtylist.dat'), 'rb') as fp:
+                naughtylist = pickle.load(fp)
+        except FileNotFoundError:
+            #No assigned pairs yet
+            msg = "You must have forgotten that there is nothing to remember."
+            await client.send_message(message.channel, msg)
+            return
+        while len(naughtylist)>1:
+            santa = naughtylist.pop(0)
+            child = naughtylist.pop(0)
+            msg = "Hey " + santa.name + ", just a reminder, if you are mailing your gift, it should be sent with enough time to arrive by December 20th.\nAlso, I'm sure you are well aware, but your secret child is " + child.name + "."
+            await client.send_message(santa, msg)
+        return
+
+    #Did someone say hungry?
+    lower = message.content.lower()
+    if "m hungry" in lower or "s hungry" in lower or "e hungry" in lower or "y hungry" in lower:
```

Comment:
This should probably be an `elif` instead, since the whole script is basically one giant if statement. We really need to re-architect this thing to be more modular...

Reply:
On second thought, this is fine. So if sanicxx does something like `!shitlist @someone thinks he is hungry` it'll have a chance to respond to that.